portfolio/src/main/java/com/google/sps/servlets/HistoryCommentsServlet.java

```diff
@@ -0,0 +1,166 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.sps.data.Message;
+import java.util.ArrayList;
+import java.io.IOException;
+import java.lang.Iterable;
+import java.util.Enumeration;
+import java.util.stream.Collectors;
+import java.util.Scanner;
+import org.json.*;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Cursor;
+import com.google.appengine.api.datastore.*;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/previous-comments")
+public class HistoryCommentsServlet extends HttpServlet {
+
+    private ArrayList<Message> messages;
+    private Cursor cursor;
+
+    @Override
+    public void init() {
+        this.messages = new ArrayList<Message>();
```

Comment:
same thing here, don't make `messages` an instance variable

Reply:
I wanted to know more the differences between using the Stream and Collection vs. my previous for in loop, considering that the Stream seemed a lot more complicated, including converting data types using StreamSupport and Collectors.tolist(). By studying purposes I wanted to see time and space complexity.

Reply:
I thought keeping the code simple is a good design practice and the for in loop requires only one line of code for the loop.